### PR TITLE
lighthouse: add `package` option to service

### PR DESCRIPTION
**Pull request by rolfschr**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by rolfschr** [comment]
@ritave Do you mind having a look into this?

**Comment by centromere** [comment]
@rolfschr LGTM.

**Comment by rolfschr** [comment]
Can we merge this?

**Review by gileri — APPROVED** [review]

File: nixos/modules/services/blockchain/ethereum/lighthouse.nix [hunk 1]
```diff
@@ -207,12 +207,19 @@ in {
         default = "";
         example = "";
       };
+
+      package = mkOption {
```
Comment (K900) [inline nixos/modules/services/blockchain/ethereum/lighthouse.nix:211]: mkPackageOption will be nicer here

File: nixos/modules/services/blockchain/ethereum/lighthouse.nix [hunk 2]
```diff
@@ -229,7 +236,7 @@ in {
         # make sure the chain data directory is created on first run
         mkdir -p ${cfg.beacon.dataDir}/${cfg.network}
 
-        ${pkgs.lighthouse}/bin/lighthouse beacon_node \
+        ${cfg.package}/bin/lighthouse beacon_node \
```
Comment (Frontear) [inline nixos/modules/services/blockchain/ethereum/lighthouse.nix:239]:
```suggestion
        ${lib.getExe cfg.package} beacon_node \
```

File: nixos/modules/services/blockchain/ethereum/lighthouse.nix [hunk 3]
```diff
@@ -277,7 +284,7 @@ in {
         # make sure the chain data directory is created on first run
         mkdir -p ${cfg.validator.dataDir}/${cfg.network}
 
-        ${pkgs.lighthouse}/bin/lighthouse validator_client \
+        ${cfg.package}/bin/lighthouse validator_client \
```
Comment (Frontear) [inline nixos/modules/services/blockchain/ethereum/lighthouse.nix:287]:
```suggestion
        ${lib.getExe cfg.package} validator_client \
```

**Review by Frontear — CHANGES_REQUESTED** [review]
Using `lib.getExe` is greatly preferred as it makes any possible changes forwards-compatible, otherwise great job, hopefully we can get this merged asap.

**Comment by rolfschr** [comment]
Okay, thanks for the additinal input. I updated the branch accordingly and aslo applied `nixfmt`. I'm happy to manipulate the git history upon request.

**Comment by rolfschr** [comment]
Ah, I realize something is wrong (it doesn't build). I had not run anything locally. Will look into this :shrug: .

**Comment by rolfschr** [comment]
Fixed my syntax error

```diff
diff --git a/nixos/modules/services/blockchain/ethereum/lighthouse.nix b/nixos/modules/services/blockchain/ethereum/lighthouse.nix
index 2c5e3d2bff7f..247174028adc 100644
--- a/nixos/modules/services/blockchain/ethereum/lighthouse.nix
+++ b/nixos/modules/services/blockchain/ethereum/lighthouse.nix
@@ -216,7 +216,7 @@ in
         example = "";
       };

-      package = lib.mkPackageOption pkgs "lighthouse";
+      package = lib.mkPackageOption pkgs "lighthouse" { };
     };
   };

```

**Review by r-vdp — APPROVED** [review]

**Pull request merged**